openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -298,6 +300,19 @@ def _handle_successful_authentication_and_login(user, request):
         django_login(request, user)
         request.session.set_expiry(604800 * 4)
         log.debug("Setting user session expiry to 4 weeks")
+
+        # Announce user's login
+        SESSION_LOGIN_COMPLETED.send_event(
+            user=UserData(
+                pii=UserPersonalData(
+                    username=user.username,
+                    email=user.email,
+                    name=user.profile.name if hasattr(user, "profile") else "",
```

Comment:
Hey @felipemontoya, I'm conflicted about this line. Some tests were failing when using `user.profile.name`, specifically these:
https://build.testeng.edx.org/job/edx-platform-python-pipeline-pr/33620/#showFailuresLink

More in detail:
https://build.testeng.edx.org/job/edx-platform-python-pipeline-pr/33620/testReport/junit/lms.djangoapps.course_goals.tests.test_goals/TestCourseGoalsAPI/Run_Tests___lms_unit___test_add_goal/

It seems that some tests use users without profiles and were crashing -obviously-. This is the best I could think of, but I didn't find any other example of accessing the user profile besides `user.profile.name`. Can you think of an alternative?

Reply:
I also went looking for other instances of `user.profile.name` and their tests. They end up using something like a UserFactory (https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/tests/factories.py#L77) which guarantees that the profile attr exists.

I would go with lets fix those tests as they use something like `self.user = User.objects.create_user('john', 'lennon@thebeatles.com', 'password')` which is the reason no profile attr is there.